diagnostics-app/src/app/diagnostics/diagnostics-card/diagnostics-card.component.ts

```diff
@@ -0,0 +1,178 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Defines the Card component for diagnostics.
+ * Imported by diagnostics.module.ts
+ */
+
+import { Component, Input, OnInit, OnDestroy } from '@angular/core';
+import { KeyValue } from '@angular/common';
+
+import {
+  DiagnosticsCardState,
+  DiagnosticsParams,
+  GetRoutineUpdateResponse,
+  requireResumeRoutines,
+  RoutineStatus,
+  RoutineType,
+} from '@common/dpsl'
+import {
+  DiagnosticsResponse,
+  ResponseErrorInfoMessage,
+  RoutineUpdateResponse,
+} from '@common/message';
+import { DiagnosticsService } from 'src/app/core/services/diagnostics.service';
+import { defaultDiagnosticsRefreshInterval, refreshIntervals } from '../../core/config/data-refresh-intervals';
+import { diagnosticsParams } from '../../core/config/diagnostics-params';
+
+@Component({
+  selector: 'app-diagnostics-card',
+  templateUrl: './diagnostics-card.component.html',
+  styleUrls: ['./diagnostics-card.component.css']
+})
+export class DiagnosticsCardComponent implements OnInit, OnDestroy {
+  @Input({ required: true }) routine!: RoutineType;
+
+  private _error?: ResponseErrorInfoMessage; // the error message received, undefined if no error occurs
+  private _intervalId?: number; // the interval id, undefined if there is no interval id
+  private _params?: DiagnosticsParams; // the parameter for starting a routine
+  private _routineId?: number; // the routine id of the routine
```

Comment:
nit: `// the routine id of the running routine`

Reply:
Done.